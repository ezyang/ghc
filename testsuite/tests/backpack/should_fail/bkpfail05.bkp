package h where
    signature H where
        data T = T1
package t-impl where
    module T where
        data T = T2
package p where
    include h
    include t-impl
    -- Known bug: GHC will not eagerly report an error here although
    -- it could, if it more aggressively checked for type-compatibility
    -- when a hole gets resolved
    signature H(T(..)) where
        import T
package q where
    include t-impl
    module H(T(..)) where
        import T
    -- Fortunately, you'll never be able to instantiate these signatures;
    -- it's just an unsatisfiable set of constraints.
    include p
