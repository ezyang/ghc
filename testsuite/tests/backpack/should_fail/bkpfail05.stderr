[1 of 5] Processing h
unit h(hole:H)
  [1 of 1] Compiling H[sig]           ( h/H.hsig, nothing )
[2 of 5] Processing t-impl
  Instantiating t-impl
unit t-impl
  [1 of 1] Compiling T                ( t-impl/T.hs, bkpfail05-out/t-impl/T.o )
[3 of 5] Processing p
unit p(hole:H)
- include t-impl [(T, t-impl:T)]
- include h(hole:H) []
  [1 of 1] Compiling H[sig]           ( p/H.hsig, nothing )
[4 of 5] Processing h-impl
  Instantiating h-impl
unit h-impl
- include t-impl [(T, t-impl:T)]
  [1 of 1] Including t-impl
  [1 of 1] Compiling H                ( h-impl/H.hs, bkpfail05-out/h-impl/H.o )
[5 of 5] Processing q
  Instantiating q
unit q
- include h-impl [(H, h-impl:H)]
- include p(h-impl:H) []
  [1 of 2] Including h-impl
  [2 of 2] Including p
    Instantiating p(h-impl:H)
unit p(h-impl:H)
- include t-impl [(T, t-impl:T)]
- include h(hole:H) []
    [1 of 2] Including t-impl
    [2 of 2] Including h
      Instantiating h(h-impl:H)
unit h(h-impl:H)
      [1 of 1] Compiling H[sig]           ( h/H.hsig, nothing )

bkpfail05.bkp:2:5: error: Not in scope: data constructor ‘T1’
